src/status_im2/common/device_permissions.cljs

```diff
@@ -0,0 +1,18 @@
+(ns status-im2.common.device-permissions
+  (:require
+    [quo2.foundations.colors :as colors]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn camera
+  [callback]
+  (rf/dispatch
+   [:request-permissions
+    {:permissions [:camera]
+     :on-allowed  callback
+     :on-denied   #(rf/dispatch
+                    [:toasts/upsert
+                     {:icon           :i/info
+                      :icon-color     colors/danger-50
+                      :override-theme :light
```

Comment:
do we need override-theme here? @smohamedjavid any idea what can be done so it's not needed here? 🤔

Reply:
In the **Sign-In** screen, the toast should use the `dark` theme. 
  
I just checked the **Sign-In** screen by removing the `override-theme` key here. It works as expected on both light and dark modes. 👍 

But, the `toast` component is not refactored yet to use the **theme** from the **theme-provider** context. 🤔 It's fishy. 

https://github.com/status-im/status-mobile/blob/bb3b3061c63719450f84b6d9e31531a9c881c0e4/src/status_im2/common/toasts/view.cljs#L51-L56

https://github.com/status-im/status-mobile/blob/bb3b3061c63719450f84b6d9e31531a9c881c0e4/src/quo2/components/notifications/toast/view.cljs#L79

Reply:
@J-Son89 - I'm not sure we can inject the screen's theme context in the `toast` component, as `toast` is rendered in a separate navigation overlay. 

https://github.com/status-im/status-mobile/blob/bb3b3061c63719450f84b6d9e31531a9c881c0e4/src/status_im2/navigation/core.cljs#L224-L230

To use the theme context of the screen, it needs to be rendered from the screen. 


For screens, we had the (override) theme (if required) defined in the `screens.cljs` which we utilised for the context provider

https://github.com/status-im/status-mobile/blob/bb3b3061c63719450f84b6d9e31531a9c881c0e4/src/status_im2/navigation/view.cljs#L61-L66

But `toasts` are different, we show a toast from any screen. I think `toast` will be an exception to using the `override-theme` key.

Let's have the `override-theme` key for now. Meanwhile, We will try and see if we can access the screen's theme context from the `toast` component.

Reply:
Sure this sounds fine, thanks for checking. maybe we can just use key ‘theme’ anyway? (With the exact same effect you’re describing here)